apps/backend/core/configuration/services.py

```
@@ -0,0 +1,278 @@
+
+    prometheus_url: str | None = Field(
+        default=None, description="Prometheus metrics URL (optional)"
+    )
+
+    # ========== Validators ==========
+    @field_validator("openai_api_base", "anthropic_api_base", "clerk_api_url", "roblox_api_base")
+    @classmethod
+    def validate_https_url(cls, v: str) -> str:
+        """Ensure API base URLs use HTTPS (except localhost)."""
+        if not v.startswith(("https://", "http://localhost", "http://127.0.0.1")):
+            raise ValueError(f"API base URL must use HTTPS: {v}")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)